### DISCUSSION TITLE
Typo on page 556

### DISCUSSION THREAD

**Discussion by anneleonore** [rasbt/machine-learning-book#197] [General]
Hi Sebastian,

Thank you for putting together such an informative book - I find it an enjoyable read and extremely helpful in my research.

Small note: On page 556 you are referring to **multihead_keys[2, 1]** and say we are accessing the second key value in the **second** attention head. Should that not be the **third** attention head?

The accompanying code is: https://github.com/rasbt/machine-learning-book/blob/main/ch16/ch16-part1-self-attention.py

Thanks again!

**Comment by rasbt** [comment]
Thanks for the note @anneleonore ! You are absolutely right, it should say "third". I just double-checked my version and it seems to be correct there:

<img width="822" alt="Screenshot 2025-01-05 at 10 26 12 AM" src="https://github.com/user-attachments/assets/1eb05361-3a5c-4606-8135-2eacf47a2d3c" />

I think this may have been fixed in the Nov , 2022 reprint I suspect you have the previous version somehow.
